core/src/test/java/org/apache/iceberg/TestMetricsTruncation.java

```diff
@@ -274,4 +274,17 @@ public void testTruncateStringMax() {
             "Test input with multiple 4 byte UTF-8 character where the first unicode character should be incremented")
         .isEqualTo(0);
   }
+
+  @Test
+  public void testTruncateStringMaxUpperBound() {
```

Comment:
Could we add these to the test above? I'm also fine if there is a specific reason to have them somewhere else but it seems like these would fit into the test above as just other examples. The test cases for +1 and MAX_CODE_POINT are there already right?

Reply:
Moved this to `testTruncateStringMax`

> The test cases for +1 and MAX_CODE_POINT are there already right?
Yes, there is already one test case for MAX_CODE_POINT